### PR TITLE
Add root navigation and home page layouts

### PR DESCRIPTION
**Pull request by austin94**
## Description

This PR includes the first UI and page layouts for the new Garden site. It includes:

- Root navigation including headers/footers
- Layouts associated with the home page

### Out of scope

This PR doesn't include any animation, production ready image/svg assets, final copy, or other fine-tuning. Please limit feedback to the overall layout and mobile-responsiveness. As we include more layouts we will need to update the hover/focus/active/selected states for all anchors as well.

We also don't have a complete idea on how we will source content for the 'Site Updates` section. This is not included at this time.

### Layouts

This PR introduces 2 common layouts for use later on:

- `MaxWidthLayout`
  - Applies a consistent max-width to content. Will be used in the Home page and 2-column layouts.
- `RootLayout`
  - Positions content with header and footer components

### Images

We are using 3 types of progressive image loading:

- The Garden "tombstone" in the search row of the home page is faded-in once loaded. Since this is above-the-fold we aren't using a base64 approximation for this image.
- The Foundation images apply an SVG trace approximation until the assets are loaded.
- The Patterns image uses a base64 approximation as it is below-the-fold.

You can view these different image states on initial page load.

### Garden News

I've created a local `garden-news-source` plugin to source articles from a `content/news.yml` file. 

@talia-eisenberg and the other designers. You can click the `View deployment` link below to view the most up-to-date view of this PR.

### CODE REVIEW

File: src/layouts/Root/components/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+    css={css`
+      display: flex;
+      flex-grow: 1;
+      align-items: center;
+      justify-content: center;
+      padding: ${p => p.theme.space.xxs};
+      height: 100%;
+    `}
+  >
+    <SearchInput ref={ref} isCompact={false} />
```
Comment (hzhu) [inline src/layouts/Root/components/Header.tsx]: Would this be the same as the following?

```
<SearchInput ref={ref} />
```

Or is passing `false` overridding something in `SearchInput`?

**Comment by hzhu** [comment]
The website looks great! I tried it on my Pixel 3 Android browser, and it looks good there.

I noticed there might be some rendering issues through BrowserStack's IE11 browser. Would you confirm if that's the case? This screenshot was from IE11 in 100% zoom level.

Screen Shot 2020-03-25 at 11 23 12 AM

Also, I noticed that on Safari, the header spacing seems to be gone compared to the other modern browsers:

### Chrome
Screen Shot 2020-03-25 at 11 27 41 AM

### Safari
Screen Shot 2020-03-25 at 11 27 20 AM

File: src/components/SEO.tsx [hunk 2]
```diff
@@ -6,11 +6,15 @@
...
 import React from 'react';
-import PropTypes from 'prop-types';
 import Helmet from 'react-helmet';
 import { useStaticQuery, graphql } from 'gatsby';
 
-function SEO({ description, lang, meta, title }) {
+const SEO: React.FC<{
+  description?: string;
+  lang?: string;
+  meta?: any[];
```
[the remark this answers is not in the document]
Comment (austin94) [reply src/components/SEO.tsx]: It looks like `HTMLMetaElement[]` works with the react-helmet type definitions. Added.

File: src/layouts/Home/components/Foundation.tsx [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+  </div>
+);
+
+export const Foundation: React.FC = () => {
+  const { contentImage, componentsImage, designImage } = useStaticQuery(
+    graphql`
+      query {
+        contentImage: file(relativePath: { eq: "home-content.jpg" }) {
+          childImageSharp {
+            fluid(maxWidth: 600, traceSVG: { background: "#03363D", color: "white" }) {
```
[the remark this answers is not in the document]
Comment (austin94) [reply src/layouts/Home/components/Foundation.tsx]: GraphQL queries do _not_ allow string interpolation it's a runtime value not available during build.

File: src/layouts/Root/components/Header.tsx (hunk 1 above)
Comment (austin94) [reply src/layouts/Root/components/Header.tsx]: Some earlier mockups had mixtures of `isCompact` and standard inputs. I've removed this prop for now.

**Comment by talia-eisenberg** [comment]
Responsiveness and layouts look good. Some styling things may be updated in the future pending going through all the feedback. Stuff that we know still needs to be sorted out:

- Outline on hero search input
- Shadow on header - does it match chrome or can it change?

**Comment by austin94** [comment]
@hzhu you found some flex-box weirdness. I've gotten the layout to now appear correctly across Firefox, Safari, and Chrome. IE11 is now laying out correctly, but the Garden logo is no longer visible.

I'm having trouble reproducing a fix in Browserstack so will look at this separately.

File: src/images/garden-icon.svg [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+<svg width="46px" height="46px" viewBox="0 0 46 46" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <!-- Generator: Sketch 62 (91390) - https://sketch.com -->
+    <title>Path</title>
+    <desc>Created with Sketch.</desc>
+    <g id="Symbols" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
+        <g id="garden-horizontal" transform="translate(-1.000000, -2.000000)" fill="#5DAE90">
+            <path d="M1.73004027,24.6616797 C1.73004027,37.096832 12.0988594,47.2164732 24.8402049,47.2164732 C37.5815504,47.2164732 47.8624982,37.096832 47.8624982,24.6616797 C47.8624982,24.6616797 47.8624982,2.10688624 47.8624982,2.10688624 C35.1211527,2.10688624 24.7523335,12.2265274 24.7523335,24.6616797 C24.7523335,12.2265274 14.3835144,2.10688624 1.64216892,2.10688624 C1.73004027,2.10688624 1.73004027,24.6616797 1.73004027,24.6616797 Z" id="Path"></path>
+        </g>
+    </g>
+</svg>
```
Comment (jzempel) [inline src/images/garden-icon.svg]: This should be getting sourced from `svg-icons`. (Maybe that is planned as a follow-up)

File: src/images/garden-wordmark.svg [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+                <path d="M167.931209,13.9975466 L174.348916,13.9975466 L174.348916,17.8993866 C176.453082,15.1270266 179.71454,13.4841466 183.712455,13.4841466 C191.392662,13.4841466 196.127036,18.4127866 196.127036,26.4218267 L196.127036,43.7747469 L189.498912,43.7747469 L189.498912,26.9352267 C189.498912,22.0065867 186.973913,18.9261866 182.134331,18.9261866 C177.715582,18.9261866 174.348916,22.1092667 174.348916,27.3459467 L174.348916,43.6720669 L167.826,43.6720669 L167.826,13.9975466 L167.931209,13.9975466 Z" id="Path"></path>
+            </g>
+        </g>
+    </g>
+</svg>
```
Comment (jzempel) [inline src/images/garden-wordmark.svg]: This should be getting sourced from `svg-icons`. (Maybe that is planned as a follow-up)

File: src/layouts/Home/components/Foundation.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (jzempel) [reply src/layouts/Home/components/Foundation.tsx]: Color values should be getting pulled from the `PALETTE` object of `react-theming`. No explicit hex or web values.

File: src/layouts/Home/components/News.tsx [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+      `}
+    >
+      <div
+        css={css`
+          position: absolute;
+          top: 0;
+          right: 50%;
+          bottom: 0;
+          left: 0;
+          background-color: ${p => getColor('green', 200, p.theme)};
```
Comment (jzempel) [inline src/layouts/Home/components/News.tsx]: This color really isn't theme-able. It should be getting pulled from `PALETTE`.

File: src/layouts/Home/components/Patterns.tsx [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+            lg={7}
+            order={1}
+            orderMd={0}
+            css={css`
+              @media (max-width: ${p => p.theme.breakpoints.md}) {
+                margin-bottom: ${p => p.theme.space.lg};
+              }
+            `}
+          >
+            <Img fluid={patternsImage.childImageSharp.fluid} alt="Garden patterns " />
```
Comment (jzempel) [inline src/layouts/Home/components/Patterns.tsx]: Remove trailing "` `" from alt.

File: src/components/SEO.tsx [hunk 8]
```diff
@@ -6,11 +6,15 @@
...
 import Helmet from 'react-helmet';
 import { useStaticQuery, graphql } from 'gatsby';
 
-function SEO({ description, lang, meta, title }) {
+const SEO: React.FC<{
+  description?: string;
+  lang?: string;
+  meta?: HTMLMetaElement[];
+  title: string;
+}> = ({ description, lang, meta, title }) => {
```
[the remark this answers is not in the document]
Comment (austin94) [reply src/components/SEO.tsx]: I've updated the SEO component to handle the root page and a new `titleTemplate` using the `/` separator.

File: src/images/garden-icon.svg (hunk 4 above)
Comment (austin94) [reply src/images/garden-icon.svg]: Didn't realize we had a 26px variant of this! I've moved to importing it from there.

File: src/images/garden-wordmark.svg (hunk 5 above)
Comment (austin94) [reply src/images/garden-wordmark.svg]: Didn't realize the wordmark was already added. Now provided from svg-icons.

**Comment by austin94** [comment]
> Color values should be getting pulled from the PALETTE object of react-theming. No explicit hex or web values.

From @hzhu 's comment as well, this is not possible with static GraphQL queries in Gatsby. We are unable to use string interpolation or use any PALETTE or DEFAULT_THEME values as that is a runtime value not available during site build..

**Comment by austin94** [comment]
> anchor components

For internal (relative) links we have to use the `Link` component provided by Gatsby to handle the progressive loading. This has the default `<a>` styling provided by Bedrock so we should be ok visually.

We can use the Garden `Anchor` throughout the markdown content and the table of contents for non-page links.

> typography elements

In the initial designs the font-sizes and line-heights didn't align with our typography elements. Now that they're more consistent I'll add them in.

**Comment by jzempel** [comment]
@austin94 be sure to use future design-theming mismatches (i.e. `Typography`) as an opportunity to discuss with the designer. If the mismatch is intentional, then the theming for the website should be updated accordingly and Garden used throughout. :pray:

File: src/layouts/Home/components/Foundation.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (jzempel) [reply src/layouts/Home/components/Foundation.tsx]: If I'm reading right, looks like this is a TBD feature of Gatsby: https://github.com/gatsbyjs/gatsby/issues/10482#issuecomment-570217789

**Comment by jzempel** [comment]
@austin94, bummer – netlify isn't respecting the fresshh deployment https://5e7d0465fdaaa602365f4f4c--zendeskgarden.netlify.com/

File: utils/gatsby-theme.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  };
+
+  return (
+    <ThemeProvider focus={focusVisibleRef} theme={theme}>
+      <div
+        ref={focusVisibleRef}
+        css={`
+          width: 100%;
+          height: 100%;
+        `}
```
Comment (jzempel) [inline utils/gatsby-theme.js]: This makes me wonder if we  have a forwardRef or prop spreading problem on Garden's `ThemeProvider`.

File: src/layouts/Home/components/News.tsx [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+      />
+      <MaxWidthLayout>
+        <Grid gutters="lg">
+          <Row>
+            <Col
+              sm={12}
+              md={6}
+              lg={4}
+              css={css`
+                background-color: ${p => getColor('green', 200, p.theme)};
```
[the remark this answers is not in the document]
Comment (jzempel) [reply src/layouts/Home/components/News.tsx]: Oops, stale comment.

File: utils/gatsby-theme.js (hunk 9 above)
Comment (austin94) [reply utils/gatsby-theme.js]: This is a unique scenario due to our `focus-visible` hook. Some more traditional ways would be to apply styling to the parent of this `<div>`, but the Gatsby abstractions make this difficult to do. For a majority of consumers (`create-react-app` based) it should be easier.

**Pull request merged**